docs/spec_draft.md

```diff
@@ -1061,6 +1099,74 @@ tensor and produces a `result` tensor.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.custom_call
+
+### Semantics
+
+Invokes exteral code with `inputs` as arguments and produces `results`. The
+exact mechanism is backend-specific. The inputs are described as follows:
+  * `inputs` are passed as agruments to the external API.
+  * `call_target_name` may used as an identifier for the external code.
```

Comment:
Documentation on `call_target_name` is inconsistent:

XLA:
https://www.tensorflow.org/xla/operation_semantics
target_name - Name of the function. A call instruction will be emitted which targets this symbol name.

MHLO:
https://www.tensorflow.org/mlir/hlo_ops#mhlocustom_call_mlirmhlocustomcallop
`call_target_name` and `backend_config` can be arbitrary strings, but `call_target_name` should be short as it may be used in labels.

Lets take a stance on whether it is a function name or an arbitrary string if possible.

Reply:
PR openxla/xla#159 is exploring using a call target that includes dots. The grammar of valid call_target_names is greater than that of C functions. It may be worthwhile to restrict this to valid MLIR identifiers:

```
// Identifiers
bare-id ::= (letter|[_]) (letter|digit|[_$.])*
```

AFAIK this wouldn't break anything existing. I am also on board with us documenting "arbitrary strings".

Reply:
Let's use arbitrary strings for now and figure out the rest in the extensibility RFC next year?